vflow/vset.py

```diff
@@ -179,6 +179,13 @@ def predict_proba(self, *args):
             if hasattr(v, 'predict_proba'):
                 pred_dict[k] = v.predict_proba
         return self._apply_func(pred_dict, *args)
+    
+    def predict_with_uncertainties(self, *args, **kwargs):
```

Comment:
Do you mean the computed mean/std should be replicated for each key in the output of `predict_proba` or that there should be a key for each row of the `pred_stats` output?

Reply:
Ya, sorry, I meant the latter.

Actually, it will be better to not use `wrt` and instead add an arg `groups` (also a list) which we pass directly to `perturbation_stats`. We have to be sure the averages are over predictions from matching input data otherwise the output is meaningless, so something like `groups = ['subsampling']` will be common.

Reply:
A few more thoughts:
- It's probably better to return two separate dicts, one for the means and one for the stds, in case we want to go on to use the mean predictions later on.
- This method could just as easily use `predict` instead of `predict_proba`, so maybe we should add a `with_uncertainty=False` arg to `predict` and `predict_proba` and have them call this method (then input to this method would be the predictions dict).
- This is also getting us closer to ensembling: right now we just return an equally-weighted average, by why not do a weighted average based on the training accuracy for each of the groups? Or via some other ensembling scheme?

Reply:
I like the `with_uncertainty` idea but we would either need to add optional `group_by` and `wrt_col` args to both `predict` and `predict_proba` or force users to call
```python
preds = predict_proba(X_test, with_uncertainty=False)
mean_preds, std_preds = predict_with_uncertainties(preds, group_by=[...], wrt_col='...') 
```
if they want more control over grouping, which seems a little counterintuitive. I just pushed most of the updates you had mentioned (with the exception of the dict wrapper and `with_uncertainty`). Assuming we want to use these dicts in future calls to some Vset I need to recreate the subkeys as `Subkey`s - I can get info for `Subkey.value` and `Subkey.origin` but not `_output_matching` and the `_sep_dicts_id` is lost since the object is new, though I'm not sure thats really important. Do you think that will eventually break?

Reply:
@rushk014 ya I think this could become an issue, though I can't come up with anything off of the top of my head right now. But some changes I have in mind to Subkeys and dict wrapping may solve the problem. For now, can you try create a reproducible example and include it in a new issue?